### PR TITLE
TapeStation Tube File IDs

### PR DESCRIPTION
**Pull request by sdjmchattie**
This is work on top of the branch James was working on so I've set the target to that.  If preferred, we can change the destination to `devel` but I thought this would be easier to review like I've set it up.

### CODE REVIEW

**Review by JamesGlover — CHANGES_REQUESTED** [review]

File: src/components/QuantFile.vue [hunk 1]
```diff
@@ -52,31 +53,70 @@ export default {
...
+      )
+      if (!fileNameMatch) {
+        return null
+      }
+
+      const groups = fileNameMatch.groups
+      const barcode = this.quantType.fileNameSpecs.barcodeFormat.replace(
+        /{(.+?)}/g,
+        function (match, key) {
+          return typeof groups[key] != 'undefined' ? groups[key] : match
```
[the remark this answers is not in the document]
Comment (sdjmchattie) [reply src/components/QuantFile.vue:76]: It's a bit of a fail safe to be honest.  It's not the matches we're passing to the regular expression groups object.  It's the name in curly braces from the barcode format.  So in this case `{yy}{mm}{dd}-{HH}{MM}{SS}`.  If `yy` were not included as a named group in the regular expression, the word `undefined` would appear in the barcode.  The line you've highlighted here would leave `{yy}` as it is written in the barcode format, in that situation.

File: tests/unit/QuantFile.spec.js [hunk 2]
```diff
@@ -356,32 +356,100 @@ describe('QuantFile.vue', () => {
     })
   })
 
-  describe('custom plate sizes', () => {
+  describe('Heron TapeStation Tubes', () => {
```
Comment (JamesGlover) [inline tests/unit/QuantFile.spec.js:359]: I did this specifically as we should be testing the behaviour of QuantFile here, not our configuration. I realise most the other tests are coupled to configuration, so perhaps consistency is better.

File: src/components/QuantFile.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (JamesGlover) [reply src/components/QuantFile.vue:76]: Is it worth the added complexity to handle what would be a configuration error (or I suppose a possible future scenario where we have filenames containing {})? Both formats of output seem to be as informative as each other.

[the remark this answers is not in the document]
Comment (sdjmchattie) [reply src/components/QuantFile.vue]: It's not about filenames containing curly braces, it's about barcode formats containing curly braces.  The barcode format here is specified as `"{yy}{mm}{dd}-{HH}{MM}{SS}"` but if you wanted to include curly braces in the generated barcodes; let's say you wanted the generate barcode to be `211021-094100-{Heron}` then setting the format to `{yy}{mm}{dd}-{HH}{MM}{SS}-{Heron}` in the format would instead generate the barcode `211021-094100-undefined`.

**Pull request merged**